### PR TITLE
[core] Support to query indexes

### PR DESCRIPTION
**Pull request by yuzelin**
### Purpose

Provide a convienient way to query index files

### CODE REVIEW

File: docs/content/concepts/system-tables.md [hunk 1]
```diff
@@ -389,6 +389,25 @@ SELECT * FROM T$statistics;
 */
 ```
 
+### Indexes Table
+
+You can query the table's index files generated for dynamic bucket table (index_type = HASH) and deletion vectors
+(index_type = DELETION_VECTORS) through indexes table.
+
+```sql
+SELECT * FROM my_table$indexes;
```
Comment (JingsongLi) [inline docs/content/concepts/system-tables.md:398]: my_table$table_indexes

File: paimon-core/src/main/java/org/apache/paimon/table/system/IndexesTable.java [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.paimon.catalog.Catalog.SYSTEM_TABLE_SPLITTER;
+import static org.apache.paimon.utils.SerializationUtils.newStringType;
+
+/** A {@link Table} for showing committing snapshots of table. */
```
Comment (xuzifu666) [inline paimon-core/src/main/java/org/apache/paimon/table/system/IndexesTable.java:67]: comments not right? showing indexes of table

**Comment by wwj6591812** [comment]
+1

**Comment by wwj6591812** [comment]
+1

**Comment by JingsongLi** [comment]
+1

**Pull request merged**